File: sdks/ruby/examples/workflow_pause/pause.rb
# frozen_string_literal: true

require "hatchet-sdk"

hatchet = Hatchet::Client.new

pausable_workflow = hatchet.workflow(name: "PausableWorkflow")

# > Pause a workflow
pausable_workflow.pause(
  queue_ttl: "24h",
  paused_workflow_cron_run_queue_behavior: "DROP",
  paused_workflow_scheduled_run_queue_behavior: "QUEUE"
)
# !!

# > Unpause a workflow
pausable_workflow.unpause
# !!

File: sdks/python/examples/workflow_pause/pause.py
from datetime import timedelta

from examples.workflow_pause.worker import pausable_workflow

# > Pause a workflow
pausable_workflow.pause(
    queue_ttl=timedelta(hours=24),
    paused_workflow_cron_run_queue_behavior="DROP",
    paused_workflow_scheduled_run_queue_behavior="QUEUE",
)
# !!

# > Unpause a workflow
pausable_workflow.unpause()
# !!
